python/cudf/cudf/core/column/numerical.py

```diff
@@ -254,7 +254,13 @@ def sum_of_squares(self, dtype=None):
         return libcudf.reduce.reduce("sum_of_squares", self, dtype=dtype)
 
     def round(self, decimals=0):
-        data = Buffer(cudautils.apply_round(self.data.mem, decimals))
+        if decimals < 0:
+            msg = "Decimal values < 0 are not yet supported."
+            raise NotImplementedError(msg)
+
+        data = Buffer(
+            cudautils.apply_round(self.astype("float").data.mem, decimals)
```

Comment:
Why do we need to typecast before handing to the numba kernel here?

Reply:
Typecasting makes the round logic easier. Happy to revisit this as it does come with a tradeoff, but I think it's worth it.

>Fixes integer handling by abstracting away the issue stemming from integer-related floating point floor division/modulo by explicitly casting to float dtype. It's fast enough already that this tradeoff for rounding integers is worthwhile, especially since this is not commonly used on integer columns.

Reply:
I'd be fine if we asserted decimal >= 0 and then just returned `self` if dtype is integer for now and had a followup for enabling decimal < 0.

Reply:
Sounds good, thanks @kkraus14 . Will do